### PR TITLE
nixos/kerberos: port tests to python

### PR DESCRIPTION
**Pull request by kwohlfahrt**
###### Motivation for this change

Port tests for #72828 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @tfc

### CODE REVIEW

File: nixos/tests/kerberos/heimdal.nix [hunk 1]
```diff
@@ -23,31 +23,21 @@ import ../make-test.nix ({pkgs, ...}: {
...
-      "kadmin -l ext_keytab --keytab=alice.keytab alice"
-    );
-    $machine->succeed("kinit -kt alice.keytab alice");
+    machine.succeed("kadmin -l add --password=admin_pw --use-defaults admin")
+    machine.succeed("kadmin -l ext_keytab --keytab=admin.keytab admin")
+    machine.succeed(
+        "kadmin -p admin -K admin.keytab add --password=alice_pw --use-defaults alice"
+    )
+    machine.succeed("kadmin -l ext_keytab --keytab=alice.keytab alice")
+    machine.succeed("kinit -kt alice.keytab alice")
```
Comment (tfc) [inline nixos/tests/kerberos/heimdal.nix]: the whole test seems to be one big test case and not semantically unrelated test cases where individual output strings are used. Would anything speak against this slick implementation?:

```python
     machine.succeed(
        "kadmin -l init --realm-max-ticket-life='8 day' --realm-max-renewable-life='10 day' FOO.BAR",
        "systemctl restart kadmind.service kdc.service"
    )

    for unit in "kadmind.service", "kdc.service", "kpasswdd.service":
        machine.wait_for_unit(unit)

    machine.succeed(
        "kadmin -l add --password=admin_pw --use-defaults admin",
        "kadmin -l ext_keytab --keytab=admin.keytab admin",
        "kadmin -p admin -K admin.keytab add --password=alice_pw --use-defaults alice",
        "kadmin -l ext_keytab --keytab=alice.keytab alice",
        "kinit -kt alice.keytab alice"
    )
```

Comment (tfc) [reply nixos/tests/kerberos/heimdal.nix]: This is just a minor style suggestion. Not a blocker.

Comment (kwohlfahrt) [reply nixos/tests/kerberos/heimdal.nix]: I didn't know `machine.succeed` takes multiple arguments. Done.

File: nixos/tests/kerberos/heimdal.nix [hunk 2]
```diff
@@ -23,31 +23,20 @@ import ../make-test.nix ({pkgs, ...}: {
...
+    machine.succeed(
+        "kadmin -l init --realm-max-ticket-life='8 day' --realm-max-renewable-life='10 day' FOO.BAR",
+        "systemctl restart kadmind.service kdc.service",
+    )
 
-    $machine->succeed(
-      "kadmin -l init --realm-max-ticket-life='8 day' \\
-       --realm-max-renewable-life='10 day' FOO.BAR"
-    );
+    for unit in ["kadmind", "kdc", "kpasswdd"]:
```
Comment (tfc) [inline nixos/tests/kerberos/heimdal.nix]: btw there is no need to make this a list. it will work without the [ ].

**Review by tfc — APPROVED** [review]

**Comment by kwohlfahrt** [comment]
FYI the last force-push is using format strings (`f"{unit}.service"` instead of `"{}.service".format(unit)"`)

File: nixos/tests/kerberos/heimdal.nix (hunk 2 above)
Comment (kwohlfahrt) [reply nixos/tests/kerberos/heimdal.nix]: Yes, but I feel like conceptually `List[str]` is more appropriate than `Tuple[str, str, str]` here.

**Comment by flokli** [comment]
Thanks!

**Pull request merged**